packages/react-resizable-panels/src/Panel.test.tsx

```diff
@@ -907,8 +905,8 @@ describe("PanelGroup", () => {
     });
 
     it("should support sizes with many decimal places", () => {
-      let panelRef = createRef<ImperativePanelHandle>();
-      let onResize = jest.fn();
+      const onResize = jest.fn();
+      const panelRef = createRef<ImperativePanelHandle>();
```

Comment:
TBH I don't really like this ESLint rule.

Reply:
I don't _specifically_ prefer `let` over `const` but I also don't think ESLint should have an opinion on the matter. Especially for something like a React ref, which is mutable, declaring it as a "constant" is kind of misleading– because it can still be _changed_ (mutated).

Reply:
> I don't _specifically_ prefer `let` over `const` but I also don't think ESLint should have an opinion on the matter. Especially for something like a React ref, which is mutable, declaring it as a "constant" is kind of misleading– because it can still be _changed_ (mutated).

I think using `const` makes the code more readable (because we won't assume the variable is getting assigned to later on). It's not really about mutability and more about direct assignment. So like in the code when I see `let strict = false` I will just know that the value of `strict` is being assigned directly later in the code. Whereas in something like an object (e.g. React ref) we can't know it's being mutated regardless of `let` or `const`. I personally think being strict about that is a good thing.

I disabled the rule regardless 😁

Reply:
Should I also revert the `let` to `const` changes?

Reply:
> I think using `const` makes the code more readable (because we won't assume the variable is getting assigned to later on). It's not really about mutability and more about direct assignment. So like in the code when I see `let strict = false` I will just know that the value of `strict` is being assigned directly later in the code. Whereas in something like an object (e.g. React ref) we can't know it's being mutated regardless of `let` or `const`. I personally think being strict about that is a good thing.

Lines 64 to 67 of `Panel.test.tsx` is a good example of this.

```ts
describe("collapse and expand", () => {
  let leftPanelRef = createRef<ImperativePanelHandle>();
  let rightPanelRef = createRef<ImperativePanelHandle>();
  
  let mostRecentLayout: number[] | null;
  ...
}
```

Since they have `let` it will give you a good context of what's going on later on in the code. Without the ESLint rule strictly fighting against this it's impossible to know that though.

Reply:
> Without the ESLint rule strictly fighting against this it's impossible to know that though

The entire test is 25 lines long. It’s pretty quick to just look at where the variables are referenced to see how they’re being used.

I understand your preference, and I don’t think it’s crazy. I just don’t share it 🙂 and I’m the only one who maintains this project so I think it’s reasonable to stick with my own preferences.

Reply:
Oh no worries at all. I hope my comment didn't seem harsh.

My stance here is pretty subjective and I just wanted to clarify that.

> I'm just trying to help a library I like be maintained

I really appreciate this 🙇🏼